O365Troubleshooters.psm1

```diff
@@ -1138,6 +1141,339 @@ Out-File -FilePath $outputFile -InputObject $currentRecord -Encoding UTF8 -Appen
+        $TheObject | Add-Member -MemberType NoteProperty -Name Interactive -Value $Interactive
+        
+
+    if (!$LinkedToNextEntry) {
+        [string]$WriteHostTypeString = $null
+
+        $null = $Global:LogObject.Add($TheObject)
+        [int]$i = 1
+        foreach ($ObjectEntry in $Global:LogObject) {
+            if ($i -eq 1) {
+                ### For HTML
+
+                ### For CSV and Write-Host
+                Write-EntryOnScreenOrLOGFileSameOrNextLine -ObjectEntry $ObjectEntry -IsFirstEntry $true
+            }
+            else {
+                ### For HTML
+
+                ### For CSV and Write-Host
+                Write-EntryOnScreenOrLOGFileSameOrNextLine -ObjectEntry $ObjectEntry -IsFirstEntry $false
+            }
+            $i++
+        }
+
+        $Global:LogObject.Clear()
+    }
+    else {
+        $null = $Global:LogObject.Add($TheObject)
+    }
+}
+
+
+<#
+  .SYNOPSIS
+  Function that actually lists entries on the screen, or add them to the .csv file
+
+  .DESCRIPTION
+  The "Write-EntryOnScreenOrLOGFileSameOrNextLine" function is the one actually listing entries on the screen, or adding them to the .csv file
+
+  .EXAMPLE
+  Write-EntryOnScreenOrLOGFileSameOrNextLine -ObjectEntry $ObjectEntry -IsFirstEntry $true
+
+  .EXAMPLE
+  Write-EntryOnScreenOrLOGFileSameOrNextLine -ObjectEntry $ObjectEntry -IsFirstEntry $false
+#>
+function Write-EntryOnScreenOrLOGFileSameOrNextLine {
+    param (
+        [bool]$IsFirstEntry,
+        $ObjectEntry
+    )
+
+    if ($ObjectEntry.Type) {
+        if ($($ObjectEntry.Type) -eq "INFO") {
+            $WriteHostTypeString = " " * 3
+        }
+        elseif ($($ObjectEntry.Type) -eq "ERROR") {
+            $WriteHostTypeString = " " * 2
+        }
+    }
+
+    [string]$WriteHostFunctionString = $null
+    if ($($ObjectEntry.Function).length -eq 0) {
+        $WriteHostFunctionString = " " * 25
+    }
+
+    [string]$WriteHostStepString = $null
+    if ($($ObjectEntry.Step).length -eq 0) {
+        $WriteHostStepString = " " * 25
+    }
+
+    [string]$WriteHostDescriptionString = " " * (42 + $($ObjectEntry.Function + $WriteHostFunctionString).Length + $($ObjectEntry.Step + $WriteHostStepString).Length)
+
+    if ($($ObjectEntry.SameLineLikePreviousEntry)) {
+        $SameLineLikePreviousEntry = $true
+    }
+    else {
+        $SameLineLikePreviousEntry = $false
+    }
+
+    if ($($ObjectEntry.LinkedToNextEntry)) {
+        $LinkedToNextEntry = $true
+    }
+    else {
+        $LinkedToNextEntry = $false
+    }
+
+    [int]$i = 1
+    foreach ($DescriptionEntry in $($ObjectEntry.Description)) {
+        if (@($DescriptionEntry | Get-Member -MemberType NoteProperty).Count -gt 1) {
+            [System.Collections.ArrayList]$DescriptionEntryMembers = Create-DescriptionTableMembers -ListOfMembers (($DescriptionEntry | Get-Member -MemberType NoteProperty).Name) -KeyEntry $($ObjectEntry.KeyEntry)
```

Comment:
The following can be used instead of Get-Member to keep the columns order (select is mandatory to get only an object in case there is a collection of objects)
$tempobj = $obj |select -first 1
$columns = $tempobj.psobject.properties | select -ExpandProperty Name